### DISCUSSION TITLE
Proposal to Contribute: Arabic Translation for "Learn Go with Tests" 🌍

### DISCUSSION THREAD

**Discussion by youssef-abbih** [quii/learn-go-with-tests#829] [Q&A] [answered]
Hi Chris,

I want to start by expressing my gratitude for this incredible project, Learn Go with Tests. It has been an invaluable resource for me—not just for learning Go, but also for understanding and applying Test-Driven Development (TDD). Your work has truly made a positive impact on my development journey.

I’d love to give back by contributing to the project through an Arabic translation of the website. As a native Arabic speaker and someone passionate about TDD and Go, I believe this translation could help more developers in the Arabic-speaking community benefit from this resource.

Since I’m new to contributing to this repository, I’d appreciate your guidance on how to proceed. Specifically:

Does the project currently support translations or have a localization system in place?
Are there any particular guidelines or tools I should follow for this contribution?
Should I create an issue or a branch to track the progress of the translation?
Thank you again for creating such a thoughtful and practical resource. I hope I can contribute meaningfully to its growth and accessibility. Looking forward to hearing your thoughts!

Best regards,
Youssef

**Comment by quii** [comment, marked as answer]
Hi Youssef

Thanks for the kind words. 

For translations, it's pretty low-tech. Create a repo to house the translation; some have started from scratch, and others are forked. I don't mind. 

Once you are happy (and it can be a work in progress), raise a PR to add your translation to the list

https://github.com/quii/learn-go-with-tests/blob/main/README.md?plain=1#L16-L23

I only ask that you link to the original source with attribution.

![CleanShot 2024-12-24 at 09 04 40](https://github.com/user-attachments/assets/f60eb3c9-1ea0-4460-bcbc-0345656c259e)


I'd take a look at the Portuguese version; they did an excellent job. 

Good luck!
